envi/__init__.py

```diff
@@ -156,6 +156,26 @@ def archGetRegisterGroups(self):
         allr = [rname for rname in regctx.getRegisterNames()]
         return [ ('all', allr), ]
 
+    def archModifyFuncAddr(self, va, info):
+        '''
+        Can modify the VA and context based on architecture-specific info.
+        Default: return the same va, info
+
+        This hook allows an architecture to correct VA and Architecture, such 
+        as is necessary for ARM/Thumb.
+        '''
+        return va, info
```

Comment:
We really should return a copy of it.  I'm def *not* a fan of calling a function with a dict and having it modify rather than copy ( creates all kinds of troubles. imagine a caller who wanted to compare info before and after the call for some unforeseeable reason ).  Even if we *were* to modify it internally, we'd still want to return it, since any RMI/RPC mechanism would lose those changes ( ie, you'd call archMod...(foo,bar) and if that was on a remote object and the API *didn't* return the dict, the changes would *not* be inserted into your local dict, since it was serialized to the remote system and died in his function local scope. )